### PR TITLE
[feature] Added view to collect email in SAML registration

### PR DESCRIPTION
**Pull request by pandafy**
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Description of Changes

If the IdP does not include email address in SAML response, then the user will be asked for email during sign-up.

### CODE REVIEW

File: openwisp_radius/saml/urls.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ def get_saml_urls(saml_views=None):
             saml_views.AssertionConsumerServiceView.as_view(),
             name='saml2_acs',
         ),
+        path(
+            'additional-info/',
+            saml_views.LoginAdditionalInfoView.as_view(),
```
Comment (nemesifier) [inline openwisp_radius/saml/urls.py:31]: I don't love the naming here, if we are asking only for the email here, why are we calling everything "additional info"?
I think we should call this view `LoginCollectEmailView`, the URL `collect-email/` and the template `collect-email.html`.

**Review by nemesifier — COMMENTED** [review]
All looks good except the naming, see my comment below and let me know what

File: openwisp_radius/saml/urls.py (hunk 1 above)
Comment (pandafy) [reply openwisp_radius/saml/urls.py:31]: I used a generic name in-case we used this view to ask for additional information. E.g., we could ask the user to verify the name received from the IdP.

**Review by nemesifier — CHANGES_REQUESTED** [review]
Is the `fur` translation missing?

File: openwisp_radius/locale/de/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -151,3 +151,22 @@ msgstr "Session verwalten"
...
 msgstr "Neue Session WiFi gestartet"
+
+#: templates/djangosaml2/login_additional_info.html:5
+msgid "Add Your Email to Complete Your Account Setup"
+msgstr "Fügen Sie Ihre E-Mail hinzu, um die Einrichtung Ihres Kontos abzuschließen"
+
+#: templates/djangosaml2/login_additional_info.html:6
+msgid ""
+"It looks like we didn't receive your email from your identity provider.</"
+"br>Please enter your email address below to complete setting up your account."
```
[the remark this answers is not in the document]
Comment (pandafy) [reply openwisp_radius/locale/de/LC_MESSAGES/django.po:162]: I can try splitting the string in two different paragraphs. 

This [old StackOverflow post](https://stackoverflow.com/questions/3397218/translating-text-blocks-with-django-what-to-do-with-the-html) says that it's okay to have HTML tag in the string.

**Review by nemesifier — APPROVED** [review]

**Pull request merged**